The TestFelixOnGateway tests are failing frequently

I've spotted a number of failures with the TestFelixOnGateway test.  For example:

https://semaphoreci.com/calico/calicoctl/branches/master/builds/1188

It looks like the test might be a little flaky.  Note, though that I've only spotted the flakiness since fixing the STs to actually use the locally built version of calico/node.  Recently, when we moved to using quay.io as the default image, we had unknowingly "broken" the STs so that the calico/node would always be pulled from quay.

Upshot is twofold:
-  The STs would have been using the latest release of calico/node rather than the version built for the branch/master
-  The STs would have been slower because the image would end up being pulled every time we start calico/node.  This could potentially reduce some timing windows.

Not sure which of these might be the cause.  Or perhaps something else entirely.

======================================================================
FAIL: test_egress_policy_can_block_through_traffic (tests.st.policy.test_felix_gateway.TestFelixOnGateway)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/tests/st/policy/test_felix_gateway.py", line 227, in test_egress_policy_can_block_through_traffic
    retry_until_success(self.assert_host_can_not_curl_ext, 3)
  File "/code/tests/st/utils/utils.py", line 118, in retry_until_success
    result = function()
  File "/code/tests/st/policy/test_felix_gateway.py", line 429, in assert_host_can_not_curl_ext
    self.fail("Internal host can curl external server IP: %s" % self.ext_server_ip)
AssertionError: Internal host can curl external server IP: 172.19.0.2

--- a/Makefile.calico-node
+++ b/Makefile.calico-node
@@ -10,7 +10,7 @@ CALICOCONTAINERS_VERSION?=$(shell git describe --tags --dirty --always)
 # Subcomponent versions:
 BIRD_VER := v0.2.0
 GOBGPD_VER := v0.1.2
-FELIX_VER := 2.1.0-rc2
+FELIX_VER := 2.1.0-rc3
 LIBNETWORK_PLUGIN_VER := v1.1.0
 CONFD_VER := v0.10.0-scale
 SYSTEMTEST_CONTAINER_VER := latest

--- a/Makefile.calicoctl
+++ b/Makefile.calicoctl
@@ -12,7 +12,7 @@ ARCH := amd64
 GO_BUILD_VER:=latest
 # Set libcalico-go version in glide.yaml first then do a `glide up -v` to update glide.lock.
 # Setting LIBCALICOGO_VER in the Makefile won't update the glide dependencies.
-LIBCALICOGO_VER := v1.0.2
+LIBCALICOGO_VER := v1.0.3
 ###############################################################################
 
 CALICOCTL_DIR=calicoctl

--- a/glide.lock
+++ b/glide.lock
@@ -1,5 +1,5 @@
-hash: a2efc58bdabffb6e1e2783dcff84ee5507fcb978c604fd72542c9c7258c8062e
-updated: 2017-02-01T17:11:27.089870478Z
+hash: e821ee1612764ee74f4987671e20a2cc965014b6719c5532520fc9e8d76323c8
+updated: 2017-02-07T15:24:17.271940076Z
 imports:
 - name: cloud.google.com/go
   version: 3b1ae45394a234c385be014e9a488f2bb6eef821
@@ -11,7 +11,7 @@ imports:
 - name: github.com/blang/semver
   version: 31b736133b98f26d5e078ec9eb591666edfd091f
 - name: github.com/coreos/etcd
-  version: 797d82611775fa0ac7a8da79e0e41e2b042ededf
+  version: 739accc242e0bdc963d0fc13785989e323fc3793
   subpackages:
   - client
   - pkg/fileutil
@@ -19,6 +19,7 @@ imports:
   - pkg/tlsutil
   - pkg/transport
   - pkg/types
+  - version
 - name: github.com/coreos/go-oidc
   version: 5644a2f50e2d2d5ba0b474bc5bc55fea1925936d
   subpackages:
@@ -27,6 +28,10 @@ imports:
   - key
   - oauth2
   - oidc
+- name: github.com/coreos/go-semver
+  version: 568e959cd89871e61434c1143528d9162da89ef2
+  subpackages:
+  - semver
 - name: github.com/coreos/go-systemd
   version: 48702e0da86bd25e76cfef347e2adeb434a0d0a6
   subpackages:
@@ -61,7 +66,7 @@ imports:
   - log
   - swagger
 - name: github.com/fsnotify/fsnotify
-  version: fd9ec7deca8bf46ecd2a795baaacf2b3a9be1197
+  version: a904159b9206978bb6d53fcc7a769e5cd726c737
 - name: github.com/ghodss/yaml
   version: 73d445a93680fa1a78ae23a5839bad48f32ba1ee
 - name: github.com/go-openapi/jsonpointer
@@ -87,7 +92,7 @@ imports:
 - name: github.com/google/gofuzz
   version: bbcb9da2d746f8bdbd6a936686a0a6067ada0ec5
 - name: github.com/hashicorp/hcl
-  version: 37ab263305aaeb501a60eb16863e808d426e37f2
+  version: 372e8ddaa16fd67e371e9323807d056b799360af
   subpackages:
   - hcl/ast
   - hcl/parser
@@ -102,7 +107,7 @@ imports:
 - name: github.com/imdario/mergo
   version: 6633656539c1639d9d78127b7d47c622b5d7b6dc
 - name: github.com/influxdata/influxdb
-  version: 855c567c67d7fff1e2f91617c292ea20cf26e53c
+  version: 98e19f257c33ceb8e1717fa57236de473054641e
   subpackages:
   - client/v2
   - models
@@ -112,9 +117,9 @@ imports:
 - name: github.com/juju/ratelimit
   version: 77ed1c8a01217656d2080ad51981f6e99adaa177
 - name: github.com/kelseyhightower/envconfig
-  version: 5c008110b20b657eb7e005b83d0a5f6aa6bb5f4b
+  version: 8bf4bbfc795e2c7c8a5ea47b707453ed019e2ad4
 - name: github.com/magiconair/properties
-  version: 9c47895dc1ce54302908ab8a43385d1f5df2c11c
+  version: b3b15ef068fd0b17ddf408a23669f20811d194d2
 - name: github.com/mailru/easyjson
   version: d5b7844b561a7bc640052f1b935f7b800330d7e0
   subpackages:
@@ -122,17 +127,17 @@ imports:
   - jlexer
   - jwriter
 - name: github.com/mattn/go-runewidth
-  version: 737072b4e32b7a5018b4a7125da8d12de90e8045
+  version: 14207d285c6c197daabb5c9793d63e7af9ab2d50
 - name: github.com/mcuadros/go-version
   version: 257f7b9a7d87427c8d7f89469a5958d57f8abd7c
 - name: github.com/mitchellh/go-ps
   version: e2d21980687ce16e58469d98dcee92d27fbbd7fb
 - name: github.com/mitchellh/mapstructure
-  version: bfdb1a85537d60bc7e954e600c250219ea497417
+  version: db1efb556f84b25a0a13a04aad883943538ad2e0
 - name: github.com/olekukonko/tablewriter
-  version: bdcc175572fd7abece6c831e643891b9331bc9e7
+  version: febf2d34b54a69ce7530036c7503b1c9fbfdf0bb
 - name: github.com/onsi/ginkgo
-  version: 00054c0bb96fc880d4e0be1b90937fad438c5290
+  version: bb93381d543b0e5725244abe752214a110791d01
   subpackages:
   - config
   - extensions/table
@@ -169,7 +174,7 @@ imports:
 - name: github.com/pelletier/go-buffruneio
   version: df1e16fde7fc330a0ca68167c23bf7ed6ac31d6d
 - name: github.com/pelletier/go-toml
-  version: ce7be745f09fe4ff89af8e3ea744e1deabf20ee3
+  version: d1fa2118c12c44e4f5004da216d1efad10cb4924
 - name: github.com/projectcalico/go-json
   version: 6219dc7339ba20ee4c57df0a8baac62317d19cb1
   subpackages:
@@ -179,7 +184,7 @@ imports:
 - name: github.com/projectcalico/go-yaml-wrapper
   version: 598e54215bee41a19677faa4f0c32acd2a87eb56
 - name: github.com/projectcalico/libcalico-go
-  version: 645bc68c9830a90d9e72fb750497a8c03c0d0ca1
+  version: 1cb7a035e0d535c264e5c8787c7606519c835841
   subpackages:
   - lib/api
   - lib/api/unversioned
@@ -210,31 +215,31 @@ imports:
 - name: github.com/satori/go.uuid
   version: b061729afc07e77a8aa4fad0a2fd840958f1942a
 - name: github.com/Sirupsen/logrus
-  version: d26492970760ca5d33129d2d799e34be5c4782eb
+  version: c078b1e43f58d563c74cebe63c85789e76ddb627
 - name: github.com/spf13/afero
-  version: 2f30b2a92c0e5700bcfe4715891adb1f2a7a406d
+  version: 72b31426848c6ef12a7a8e216708cb0d1530f074
   subpackages:
   - mem
 - name: github.com/spf13/cast
-  version: 24b6558033ffe202bf42f0f3b870dcc798dd2ba8
+  version: d1139bab1c07d5ad390a65e7305876b3c1a8370b
 - name: github.com/spf13/jwalterweatherman
-  version: 33c24e77fb80341fe7130ee7c594256ff08ccc46
+  version: fa7ca7e836cf3a8bb4ebf799f472c12d7e903d66
 - name: github.com/spf13/pflag
   version: 08b1a584251b5b62f458943640fc8ebd4d50aaa5
 - name: github.com/spf13/viper
-  version: 651d9d916abc3c3d6a91a12549495caba5edffd2
+  version: 5ed0fc31f7f453625df314d8e66b9791e8d13003
 - name: github.com/termie/go-shutil
   version: bcacb06fecaeec8dc42af03c87c6949f4a05c74c
 - name: github.com/ugorji/go
-  version: 9c7f9b7a2bc3a520f7c7b30b34b7f85f47fe27b6
+  version: ded73eae5db7e7a0ef6f55aace87a2873c5d2b74
   subpackages:
   - codec
 - name: github.com/vishvananda/netlink
-  version: 7a823e360eee9619b585d41579dfd445a8f788cb
+  version: dbc72376c86301bb873c332830857e73c9f701e3
   subpackages:
   - nl
 - name: github.com/vishvananda/netns
-  version: 8ba1072b58e0c2a240eb5f6120165c7776c3e7b8
+  version: 2c9454e4fc6e2edc1a1c84e64ed3d6e662fb6991
 - name: golang.org/x/crypto
   version: 1351f936d976c60a0a48d728281922cf63eafb8d
   subpackages:
@@ -242,7 +247,7 @@ imports:
   - blowfish
   - ssh/terminal
 - name: golang.org/x/net
-  version: 6acef71eb69611914f7a30939ea9f6e194c78172
+  version: f2499483f923065a842d38eb4c7f1927e6fc6e6d
   subpackages:
   - context
   - context/ctxhttp
@@ -250,6 +255,7 @@ imports:
   - http2/hpack
   - idna
   - internal/timeseries
+  - lex/httplex
   - trace
 - name: golang.org/x/oauth2
   version: 3c3a985cb79f52a3190fbc056984415ca6763d01
@@ -259,7 +265,7 @@ imports:
   - jws
   - jwt
 - name: golang.org/x/sys
-  version: 8f0908ab3b2457e2e15403d3697c9ef5cb4b57a9
+  version: 7a6e5648d140666db5d920909e082ca00a87ba2c
   subpackages:
   - unix
 - name: golang.org/x/text
@@ -429,7 +435,7 @@ imports:
   - transport
 testImports:
 - name: github.com/onsi/gomega
-  version: f1f0f388b31eca4e2cbe7a6dd8a3a1dfddda5b1c
+  version: e85f63af4302dc0e4277c6008ecf803f1d80e4f0
   subpackages:
   - format
   - internal/assertion

--- a/glide.yaml
+++ b/glide.yaml
@@ -21,7 +21,7 @@ import:
   - json
 - package: github.com/projectcalico/go-yaml-wrapper
 - package: github.com/projectcalico/libcalico-go
-  version: v1.0.2 
+  version: v1.0.3
   subpackages:
   - lib/api
   - lib/api/unversioned

--- a/tests/st/policy/test_felix_gateway.py
+++ b/tests/st/policy/test_felix_gateway.py
@@ -202,7 +202,6 @@ class TestFelixOnGateway(TestBase):
         self.add_gateway_internal_iface()
         retry_until_success(self.assert_host_can_curl_ext, 3)
 
-    @skip("See https://github.com/projectcalico/calicoctl/issues/1492")
     def test_egress_policy_can_block_through_traffic(self):
         self.add_policy({
             'apiVersion': 'v1',
@@ -326,7 +325,6 @@ class TestFelixOnGateway(TestBase):
         ('allow', 'deny'),
         ('deny', 'allow')
     ])
-    @skip("See https://github.com/projectcalico/calicoctl/issues/1492")
     def test_conflicting_ingress_and_egress_policy(self, in_action, out_action):
         # If there is policy on the ingress and egress interface then both should
         # get applied and 'deny' should win.

